kotlin/workflow-core/src/main/java/com/squareup/workflow/StartStopWorker.kt

```diff
@@ -0,0 +1,46 @@
+package com.squareup.workflow
+
+import com.squareup.workflow.Worker.Emitter
+import kotlinx.coroutines.NonCancellable
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withContext
+
+/**
+ * [Worker] that performs some action when started and when stopped.
+ */
+abstract class StartStopWorker : Worker<Nothing> {
```

Comment:
@rjrjr @davidapgar This replaces the `onTeardown` hook we used to have.

Reply:
Except we still have it?